packages/web-api-scan-runner/src/report-generator/axe-result-to-consolidated-html-converter.ts

```diff
@@ -1,38 +1,39 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
-
-import { AxeReportParameters, ReporterFactory } from 'accessibility-insights-report';
-import { AxeResults } from 'axe-core';
+import { ReporterFactory } from 'accessibility-insights-report';
+import axe from 'axe-core';
 import { inject, injectable } from 'inversify';
-import { ReportFormat } from 'storage-documents';
+import { ReportFormat, CombinedScanResults } from 'storage-documents';
+import { CombinedReportDataConverter, ScanResultData } from 'axe-result-converter';
 import { iocTypeNames } from '../ioc-types';
-import { AxeResultConverter, ReportGenerationParams } from './axe-result-converter';
 import { htmlReportStrings } from './html-report-strings';
+import { AxeResultConverterOptions } from './axe-result-converter';
 
 @injectable()
-export class AxeResultToConsolidatedHtmlConverter implements AxeResultConverter {
+export class AxeResultToConsolidatedHtmlConverter {
     public readonly targetReportFormat: ReportFormat = 'consolidated.html';
 
-    constructor(@inject(iocTypeNames.ReporterFactory) private readonly reporterFactoryFunc: ReporterFactory) {}
+    constructor(
+        @inject(CombinedReportDataConverter) private readonly combinedReportDataConverter: CombinedReportDataConverter,
+        @inject(iocTypeNames.ReporterFactory) private readonly reporterFactoryFunc: ReporterFactory,
+        private readonly axeCore: typeof axe = axe,
+    ) {}
 
-    public convert(results: AxeResults, params: ReportGenerationParams): string {
+    public convert(combinedScanResults: CombinedScanResults, options: AxeResultConverterOptions): string {
         const reporter = this.reporterFactoryFunc();
-
-        const htmlReportParams: AxeReportParameters = {
-            results: results,
-            description: this.createDescription(results),
-            serviceName: htmlReportStrings.serviceName,
-            scanContext: {
-                pageTitle: params.pageTitle,
-            },
+        const scanResultData: ScanResultData = {
+            baseUrl: options.baseUrl ?? 'n/a',
+            basePageTitle: '', // not used
+            scanEngineName: htmlReportStrings.serviceName,
+            axeCoreVersion: this.axeCore.version,
+            browserUserAgent: options.userAgent,
+            urlCount: combinedScanResults.urlCount,
+            scanStarted: options.scanStarted,
+            scanEnded: new Date(),
```

Comment:
It seems strange to me for the scanEnded date to be determined by report generation time.. Could this be another parameter in `options`?

Reply:
it could but will not change the logic, end time is approximate value b/c start/end time on a report will not reflect an actual scan time but reporting period of consolidated report.